generate/graphql.go

```diff
@@ -0,0 +1,647 @@
+	}
+	if _, err := fmt.Fprintf(gfile, "# %s\n", disclaimer); err != nil {
+		return fmt.Errorf("failed writing GraphQL: %v", err)
+	}
+
+	if !filepath.IsAbs(schemaPath) {
+		schemaPath = filepath.Join(filepath.Dir(args.SchemaPath), schemaPath)
+	}
+	schema, err := jsonschema.SchemaFromFile(schemaPath, name)
+	if err != nil {
+		return err
+	}
+
+	// Interface.
+	var common []*goGQL
+	var commonFields int
+	if len(schema.AllOf) != 0 {
+		common = make([]*goGQL, 0, len(schema.AllOf))
+		for _, allOfSchema := range schema.AllOf {
+			baseName := strings.Split(filepath.Base(schema.AllOf[0].FromRef), ".")[0]
+			if newName, ok := args.StructNameMap[baseName]; ok {
+				baseName = newName
+			} else {
+				baseName = exportedName(baseName)
+			}
+			generated, err := newGeneratedGraphQLFile(allOfSchema, baseName, packageName, nil, args)
+			if err != nil {
+				return fmt.Errorf("failed to build generated struct: %v", err)
+			}
+			common = append(common, generated)
+		}
+		// TODO: Doesn't handle multiple AllOf properly with regard to mapping to Go type name.  Ex: `BaseLinks`.
+		obj := goGQL{
+			packageName: packageName,
+			args:        args,
+			rootStruct: &generatedGraphQLObject{
+				buildType: "type",
+				gqlExtractedField: gqlExtractedField{
+					args:           args,
+					array:          false,
+					description:    "",
+					fieldOrder:     []string{},
+					fields:         map[string]*gqlExtractedField{},
+					jsonName:       packageName,
+					jsonType:       "object",
+					name:           name,
+					requiredFields: map[string]bool{},
+				},
+			},
+			nestedStructs: map[string]*generatedGraphQLObject{},
+		}
+		if len(common) != 0 {
+			obj.rootStruct.gqlExtractedField.jsonName = exportedName(packageName)
+			name := strings.Split(filepath.Base(schema.AllOf[0].FromRef), ".")[0]
+			if newName, ok := args.StructNameMap[name]; ok {
+				name = newName
+			} else {
+				name = exportedName(name)
+			}
+			obj.rootStruct.gqlExtractedField.name = name
+			for _, com := range common {
+				commonFields += len(com.rootStruct.fields)
+				for fk, fv := range com.rootStruct.fields {
+					obj.rootStruct.fields[fk] = fv
+					obj.rootStruct.fieldOrder = append(obj.rootStruct.fieldOrder, fk)
+				}
+				for rk, rv := range com.rootStruct.requiredFields {
+					obj.rootStruct.requiredFields[rk] = rv
+				}
+				for nk, nv := range com.nestedStructs {
+					obj.nestedStructs[nk] = nv
+				}
+			}
+			if len(schema.OneOf) != 0 {
+				if newName, ok := args.GraphQLTypeNameMap[packageName]; ok {
+					obj.rootStruct.gqlExtractedField.jsonName = newName
+				} else {
+					obj.rootStruct.gqlExtractedField.jsonName = exportedName(packageName)
+				}
+				obj.rootStruct.gqlExtractedField.name = obj.rootStruct.gqlExtractedField.jsonName
+				obj.rootStruct.buildType = "interface"
+			}
+		}
+		if err := obj.write(gfile); err != nil {
+			return fmt.Errorf("failed to write allOf struct: %w", err)
+		}
+	}
+
+	// Implementations.
```

Comment:
Okay, seeing this comment makes the `Interface` comment above make more sense. But I think the comments could still be a bit more descriptive?

Reply:
Added some more detail to the comments.